File: src/unix/apple/iphoneos/system.m
// This Source Code Form is subject to the terms of the MIT License.
// If a copy of the MIT License was not distributed with this file,
// You can obtain one at https://spdx.org/licenses/MIT.html.

#include "matoya.h"

const char *MTY_GetSOExtension(void)
{
	return "dylib";
}

uint32_t MTY_GetPlatform(void)
{
	return MTY_OS_UNKNOWN;
}

uint32_t MTY_GetPlatformNoWeb(void)
{
	return MTY_GetPlatform();
}

void MTY_HandleProtocol(const char *uri, void *token)
{
}

const char *MTY_GetProcessPath(void)
{
	return "/app";
}

bool MTY_StartInProcess(const char *path, char * const *argv, const char *dir)
{
	return false;
}

bool MTY_RestartProcess(char * const *argv)
{
	return MTY_StartInProcess(NULL, argv, NULL);
}

File: src/system.c
// Copyright (c) Christopher D. Dickson <cdd@matoya.group>
//
// This Source Code Form is subject to the terms of the MIT License.
// If a copy of the MIT License was not distributed with this file,
// You can obtain one at https://spdx.org/licenses/MIT.html.

#include "matoya.h"

#include <stdio.h>

#include "tlocal.h"

const char *MTY_GetPlatformString(uint32_t platform)
{
	uint8_t major = (platform & 0xFF00) >> 8;
	uint8_t minor = platform & 0xFF;

	char *ver = mty_tlocal(16);

	if (minor > 0) {
		snprintf(ver, 16, "%u.%u", major, minor);

	} else {
		snprintf(ver, 16, "%u", major);
	}

	return ver;
}

bool MTY_IsSupported(void)
{
	uint32_t platform = MTY_GetPlatform();

	MTY_OS os = platform & 0xFF000000;
	uint32_t major = (platform & 0xFF00) >> 8;
	uint32_t minor = platform & 0xFF;

	switch (os) {
		case MTY_OS_WINDOWS:
			return major > 6 || (major == 6 && minor >= 1);
		case MTY_OS_MACOS:
			return major > 10 || (major == 10 && minor > 11);
		case MTY_OS_ANDROID:
			return minor >= 26;
		case MTY_OS_LINUX:
		case MTY_OS_WEB:
			return true;
		case MTY_OS_IOS:
		case MTY_OS_TVOS:
		default:
			return false;
	}
}
